### PR TITLE
Allow users to delete uploaded files.

### LINKED ISSUE #1793: The uploads controller should enable deleting of uploads
This will allow us to clean up some of the uploads that aren't needed. It won't help us if the user just leaves the page, so we'll still need a method for cleaning up abandoned uploads.

- [ ] Add a delete method to the uploads controller that removes the file and the row in the database
- [ ] Un-hide the `Cancel Upload` button after files are added. It has the effect of triggering a delete/cleanup of all affected files and database rows on the back-end.
- [ ] Make per-file `Cancel` buttons trigger a delete/cleanup of the appropriate file and database row on the back-end.
- [ ] Authorize that the user can delete the upload
- [ ] Send `deleteUrl` in the json response to create.
- [ ] Make the upload form call the delete path when removing:
```
            {% if (file.deleteUrl) { %}
                <button class="btn btn-danger delete" data-type="{%=file.deleteType%}" data-url="{%=file.deleteUrl%}"{% if (file.deleteWithCredentials) { %} data-xhr-fields='{"withCredentials":true}'{% } %}>
                    <i class="glyphicon glyphicon-trash"></i>
                    <span>Delete</span>
                </button>
                <input type="checkbox" name="delete" value="1" class="toggle">
            {% } else { %}
                <button class="btn btn-warning cancel">
                    <i class="glyphicon glyphicon-ban-circle"></i>
                    <span>Cancel</span>
                </button>
            {% } %}
```

### LINKED ISSUE #1910: Batch uploads controller should enable deleting of uploads
This will allow us to clean up some of the uploads that aren't needed. It won't help us if the user just leaves the page, so we'll still need a method for cleaning up abandoned uploads.

- [ ] Add a delete method to the uploads controller that removes the file and the row in the database
- [ ] Un-hide the `Cancel Upload` button after files are added. It has the effect of triggering a delete/cleanup of all affected files and database rows on the back-end.
- [ ] Make per-file `Cancel` buttons trigger a delete/cleanup of the appropriate file and database row on the back-end.
- [ ] Authorize that the user can delete the upload
- [ ] Send `deleteUrl` in the json response to create.
- [ ] Make the upload form call the delete path when removing:
```
            {% if (file.deleteUrl) { %}
                <button class="btn btn-danger delete" data-type="{%=file.deleteType%}" data-url="{%=file.deleteUrl%}"{% if (file.deleteWithCredentials) { %} data-xhr-fields='{"withCredentials":true}'{% } %}>
                    <i class="glyphicon glyphicon-trash"></i>
                    <span>Delete</span>
                </button>
                <input type="checkbox" name="delete" value="1" class="toggle">
            {% } else { %}
                <button class="btn btn-warning cancel">
                    <i class="glyphicon glyphicon-ban-circle"></i>
                    <span>Cancel</span>
                </button>
            {% } %}
```
### Related work

Refs #1793

### PR DESCRIPTION
**Pull request by jcoyne**
@projecthydra/sufia-code-reviewers

Fixes #1910
Fixes #1793

### CODE REVIEW

File: app/views/sufia/batch_uploads/_form_files.html.erb [hunk 1]
```diff
@@ -98,10 +98,17 @@
...
-              </button>
+              {% if (file.deleteUrl) { %}
+                <button class="btn btn-danger delete pull-right" data-type="{%=file.deleteType%}" data-url="{%=file.deleteUrl%}"{% if (file.deleteWithCredentials) { %} data-xhr-fields='{"withCredentials":true}'{% } %}>
+                    <i class="glyphicon glyphicon-trash"></i>
+                    <span>Delete</span>
+                </button>
+              {% } else { %}
+                <button class="btn btn-warning cancel pull-right">
+                    <i class="glyphicon glyphicon-ban-circle" aria-hidden="true"></i>
+                    <span>Cancel</span>
```
Comment (mjgiarlo) [inline app/views/sufia/batch_uploads/_form_files.html.erb]: What does cancel do?

Comment (jcoyne) [inline app/views/sufia/batch_uploads/_form_files.html.erb]: Removes the file from the Work, but doesn't remove the upload.

Comment (mjgiarlo) [inline app/views/sufia/batch_uploads/_form_files.html.erb]: Do we want that?

Comment (jcoyne) [inline app/views/sufia/batch_uploads/_form_files.html.erb]: I suppose it's dead code at the moment. I suppose I was thinking that someone may want that to be configurable sometime in the future.

Comment (mjgiarlo) [inline app/views/sufia/batch_uploads/_form_files.html.erb]: My preference is to expose only the button that triggers a back-end delete, and I don't care whether we call it Delete or Cancel. I defer to folks like @mtribone @ggeisler @aeschylus.

**Comment by jcoyne** [comment]
@mjgiarlo changes made

**Pull request merged**